core/controllers/profile.py

```diff
@@ -139,6 +153,33 @@ def post(self, _):
 class PreferencesHandler(base.BaseHandler):
     """Provides data for the preferences page."""
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'update_type': {
+                'schema': {
+                    'type': 'basestring',
+                    'choices': [
+                        'user_bio',
+                        'subject_interests',
+                        'preferred_language_codes',
+                        'preferred_site_language_code',
+                        'preferred_audio_language_code',
+                        'profile_picture_data_url',
+                        'default_dashboard',
+                        'email_preferences'
+                    ]
+                }
+            },
+            'data': {
```

Comment:
@SD-13 Can we check length of ```data``` argument and then remove the dead code ?

Reply:
@Nik-09 I have tried to do so. Please have a look . Thanks.